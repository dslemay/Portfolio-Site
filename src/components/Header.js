import React from 'react';
import Link from 'gatsby-link';
import { css } from 'emotion';
import styled from 'react-emotion';

import logo from '../assets/logo.png';

const HeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 1.25rem;
  background-color: ${props => props.theme.baseColor};
`;

const Logo = styled.img`
  margin: 0.5em 0.5em;
`;

const NavigationLinks = styled.ul`
  margin-bottom: 0;
  margin-right: 2.5em;

  a {
    color: ${props => props.theme.lightAccent};
  }
`;

const LinkStyle = css`
  margin-right: 2.5em;
  text-decoration: none;
`;

const NavLink = props => (
  <Link to={props.to} className={LinkStyle}>
    {props.text}
  </Link>
);

const Header = () => (
  <HeaderWrapper>
    <Link to="/">
      <Logo src={logo} />
    </Link>
    <NavigationLinks>
      <NavLink to="/about" text="About" />
      <NavLink to="/work" text="Work" />
      <NavLink to="/contact" text="Contact" />
      <NavLink to="/blog" text="Blog" />
    </NavigationLinks>
  </HeaderWrapper>
);

export default Header;
