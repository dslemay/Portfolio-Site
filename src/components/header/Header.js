import React, { Component } from 'react';
import { Link } from 'gatsby';
import { animateScroll as scroll } from 'react-scroll';
import styled from 'react-emotion';
import NavigationLinks from './NavigationLinks';
import MobileNavigation from './MobileNavigation';
import themeUtils from '../themeUtils';
import logo from '../../assets/img/logo.png';

const HeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 1.25rem;
  background-color: ${themeUtils.baseColor};
  z-index: 10;
  position: fixed;
  width: 100%;
  height: 65px;
  top: 0;
  left: 0;
`;

const Logo = styled.img`
  margin: 0.5em 0.5em;
`;

class Header extends Component {
  constructor() {
    super();
    this.state = {
      mobile: true,
    };
    this.handleSizeChange = this.handleSizeChange.bind(this);
  }

  componentDidMount() {
    const mql = window.matchMedia('(max-width: 450px)');
    mql.addListener(this.handleSizeChange);
    this.handleSizeChange(mql);
    this.path = window.location.pathname;
  }

  handleSizeChange(evt) {
    const { mobile } = this.state;

    if (evt.matches && !mobile) {
      this.setState({ mobile: true });
    }
    if (!evt.matches && mobile) {
      this.setState({ mobile: false });
    }
  }

  render() {
    const { mobile } = this.state;

    const home = this.path === '/';

    return (
      <HeaderWrapper>
        <Link
          to="/"
          onClick={e => {
            if (home) {
              e.preventDefault();
              scroll.scrollToTop();
            }
          }}
        >
          <Logo src={logo} />
        </Link>
        {mobile ? <MobileNavigation home={home} /> : <NavigationLinks home={home} />}
      </HeaderWrapper>
    );
  }
}

export default Header;
