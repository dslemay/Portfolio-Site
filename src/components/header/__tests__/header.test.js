import React from 'react';
import { animateScroll as scroll } from 'react-scroll';
import { cleanup, fireEvent, render } from 'react-testing-library';

import Header from '..';

afterEach(cleanup);

const location = {
  pathname: '/',
};

scroll.scrollToTop = jest.fn();

describe('<Header /> mobile view', () => {
  beforeAll(() => {
    window.matchMedia = jest.fn().mockImplementation(query => ({
      matches: true,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    }));
  });

  it('renders correctly', () => {
    const { container, getByLabelText } = render(<Header location={location} />);

    expect(container).toBeTruthy();
    expect(getByLabelText('menu')).toBeTruthy();
  });

  it('the mobile icon toggles open states on click', () => {
    const { getByLabelText } = render(<Header location={location} />);

    const menu = getByLabelText('menu');
    expect(menu).toHaveAttribute('aria-expanded', 'false');
    fireEvent.click(menu);
    expect(menu).toHaveAttribute('aria-expanded', 'true');
  });
});

describe('<Header /> non-mobile view', () => {
  beforeAll(() => {
    window.matchMedia = jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    }));
  });

  it('renders correctly', () => {
    const { container, getByAltText } = render(<Header location={location} />);

    const logo = getByAltText('Logo');
    fireEvent.click(logo);

    expect(container).toBeTruthy();
    expect(logo).toBeTruthy();
    expect(scroll.scrollToTop).toBeCalledTimes(1);
    scroll.scrollToTop.mockReset();
  });
});
