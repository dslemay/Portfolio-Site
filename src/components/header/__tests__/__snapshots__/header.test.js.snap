// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> renders correctly 1`] = `
.emotion-0 {
  margin-right: 1.5em;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline;
}

.emotion-4 {
  margin-bottom: 0;
  margin-right: 0;
  background-color: #2D3B7F;
}

.emotion-4 a {
  color: #AFB8E6;
  cursor: pointer;
  -webkit-transition: color 500ms;
  transition: color 500ms;
}

.emotion-4 a:visited {
  color: #AFB8E6;
}

.emotion-4 a:hover {
  color: #9A8956;
}

<nav
  class="emotion-4"
>
  <a
    class="emotion-0"
    href="#about"
  >
    About
  </a>
  <a
    class="emotion-0"
    href="#work"
  >
    Work
  </a>
  <a
    class="emotion-0"
    href="#contact"
  >
    Contact
  </a>
  <mockedlink
    class="emotion-0"
    to="/blog"
  >
    Blog
  </mockedlink>
</nav>
`;
